### DISCUSSION TITLE
Layer Time Smoothing

### DISCUSSION THREAD

**Discussion by DARKFiB3R** [OrcaSlicer/OrcaSlicer#6358] [Ideas]
It would be nice if there was an option that attempts to automatically make layer print times more equal (within reason), in order to give an even amount of sheen to the surface finish.

"Don't slow down outer walls" is great, but it's not always enough.

I presume this would try to adjust max volumetric speed, infill speed, overhang speed, etc. With the goal of making all layers have a much more similar print time, and also smooth out transitions between large differences where they may be necessary, while also trying to maintain a specified outer wall speed, to avoid VFAs.

EDIT: For anyone finding this page in the future, there has been some very interesting discussion taking place over here > https://github.com/SoftFever/OrcaSlicer/pull/8107

**Comment by ramorimdias** [comment]
Hello,

I came across this weird behavior and think it's exactly what you described. Take a look at this print: due to the cross-hatch infill, some parts of the print didn't have infill at all (I believe this happens when the parts align with the straight pattern of the infill). This causes slight variations in layer time, leading to differences in the surface finish of the print. The flow rate, speeds, and layer height are consistent throughout the entire print. This issue is caused solely by layer time—I didn't know that could cause such an effect.

![Captura de tela 2024-11-17 134252](https://github.com/user-attachments/assets/69dd493c-3446-459e-8aa5-6910a4d955f8)
![Captura de tela 2024-11-17 134259](https://github.com/user-attachments/assets/ec639263-1176-42d6-904d-50b3017baec8)

![WhatsApp Image 2024-11-17 at 13 45 44](https://github.com/user-attachments/assets/a826977d-21ef-4068-b4bd-3121668b4cd7)

**Reply by DARKFiB3R** [reply]
The only way I have found to really eliminate this (after making small improvments by messing many other settings), is by having 2 or more copies of the same object on the build plate, so that the fastest layers now take at least twice as long.

I suppose some other random object could also work, as long as it increases the overall layer time by a sufficient amount.

In your example, I would try experimenting with different types of infill, and adding more walls. I'm pretty sure that would help.

**Comment by TheOverpassArsonist** [comment]
I second this idea. 
<img width="1148" height="642" alt="image" src="https://github.com/user-attachments/assets/028e18a6-e4da-44c6-843f-dd7f242664f0" />
In this benchy it's almost perfect, but I get the worst artifacts wherever there is a _sharp difference_ in layer time. Not lowest layer time or highest layer time, but the sharpest difference between adjacent layers.

The naive solution is to increase the minimum layer time, but then that makes the _entire_ print slower, (SIGNIFICANTLY so) and I'm not actually having issues with fast layers because the chimney prints fine. Smoothing speed/layertimes between adjacent layers (maybe with a configurable 'blur-radius' for how many layers are smoothed together) would fix some of the artifacts on this benchy without making it take significantly longer.

Going fast is easy, but going fast _and_ high quality is where the problems come up, and as far as I'm aware the only solution to this issue that exists (that wouldn't make the entire print take much longer) is manually setting Z-height based options to smooth out the speed on those layer boundaries

**Comment by bauflob** [comment]
First time poster here.
I searched the treats as good as was able,i think, and i guess this is the best place to comment apart from opening a new feature request.

Lately i have been thinking along those lines as well and think something like that "could / should" be possible to implement. fairly easy (says someone without any coding ability whatsoever)
But for a manual change ..
So: There is already a "slider" availible for adjustable layer hight.  so thats taken care of. would "only" need to be linked to the otherwise fixed print-speed  instead of (again otherwise fixed layer hight).  Working like the layerhight  slider (ramping) would give manual control including a transition, just what some here looking for.
So only a new "link" or connection or however you people with far more skills than me call it  would need to be created.

Hope this will get picked up...

Greets and thanks to all of you creating this for us mere mortals

**Comment by abillits** [comment]
I would love a feature that ensures consistent outer wall times.

I've been able to minimize this problem by staying away from certain infill patterns. For example, here's a comparison with gyroid and rectilinear:
<img width="3840" height="2088" alt="gyroid" src="https://github.com/user-attachments/assets/cd9f922f-fbd6-442a-a521-0e80d96cf8f7" />
<img width="3840" height="2088" alt="rectilinear" src="https://github.com/user-attachments/assets/4c25988d-eb2e-4dab-a979-5e0b130ec408" />

**Reply by Incendos** [reply]
Just stumbled upon this thread and yeah, I was looking for exactly this. Unfortunately it's still not implemented... and I can't get away with different infill type, because my model has solid thin walls (Hexagon holes in the walls change the layer time drastically).

**Reply by DARKFiB3R** [reply]
I haven't been keeping track regarding any of this stuff, but I'm sure some progress has been made to some extent.

If you need a quick fix, I still think adding a second item to the build plate can help (in theory) by increasing the layer time for all layers.

I think at some point, if all layer times are long enough, they will all start to look the same? Hopefully without having to use an excessive amount extra filament or increasing the overall print time by too much 🤷🏻‍♂️

Maybe some sort of sacrificial inverse version of the actual model, where it only causes an increase in layer time by a significant amount where needed?

I don't know. Those are just some (probably flawed) ideas for consideration :)
